crates/next-api/src/module_graph.rs

```diff
@@ -285,21 +292,25 @@ impl SingleModuleGraph {
             }
         }
 
-        let root_idx = root.and_then(|root| {
-            if !modules.contains_key(&root) {
-                let root_idx = graph.add_node(SingleModuleGraphNode {
-                    module: root,
-                    issues: Default::default(),
-                    layer: None,
-                });
-                for entry in entries {
-                    graph.add_edge(root_idx, *modules.get(entry).unwrap(), ());
+        let root_idx = match root {
+            Some(root) => {
```

Comment:
I would change this to a `if let Some(root) = root {` instead of a match for readablility